### PR TITLE
Plugins: Improve wording around allow_loading_unsigned_plugins configuration

### PR DESCRIPTION
**Pull request by wbrowne**
**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana/issues/35670

### CODE REVIEW

**Review by marefr — APPROVED** [review]
LGTM 🎉

**Comment by serramatutu** [comment]
Thanks @wbrowne for fixing it and @marefr for forwarding the feedback :)

**Comment by wbrowne** [comment]
No worries @serramatutu - thanks for contributing your feedback to Grafana! ❤️

File: conf/defaults.ini [hunk 1]
```diff
@@ -887,7 +887,7 @@ disable_sanitize_html = false
 [plugins]
 enable_alpha = false
 app_tls_skip_verify_insecure = false
-# Enter a comma-separated list of plugin identifiers to identify plugins that are allowed to be loaded even if they lack a valid signature.
+# Enter a comma-separated list of plugin identifiers to identify plugins that should be loaded even if they are unsigned. Plugins with modified signatures will never be loaded.
```
Comment (achatterjee-grafana) [inline conf/defaults.ini:890]:
```suggestion
# Enter a comma-separated list of plugin identifiers to identify plugins to load even if they are unsigned. Plugins with modified signatures are never loaded.
```

File: conf/sample.ini [hunk 2]
```diff
@@ -873,7 +873,7 @@
 [plugins]
 ;enable_alpha = false
 ;app_tls_skip_verify_insecure = false
-# Enter a comma-separated list of plugin identifiers to identify plugins that are allowed to be loaded even if they lack a valid signature.
+# Enter a comma-separated list of plugin identifiers to identify plugins that should be loaded even if they are unsigned. Plugins with modified signatures will never be loaded.
```
Comment (achatterjee-grafana) [inline conf/sample.ini:876]:
```suggestion
# Enter a comma-separated list of plugin identifiers to identify plugins to load even if they are unsigned. Plugins with modified signatures are never loaded.
```

File: docs/sources/administration/configuration.md [hunk 3]
```diff
@@ -1474,7 +1474,9 @@ Set to `true` if you want to test alpha plugins that are not yet ready for gener
 
 ### allow_loading_unsigned_plugins
 
-Enter a comma-separated list of plugin identifiers to identify plugins that are allowed to be loaded even if they lack a valid signature.
+Enter a comma-separated list of plugin identifiers to identify plugins that should be loaded even if they are unsigned. Plugins with modified signatures will never be loaded.
```
Comment (achatterjee-grafana) [inline docs/sources/administration/configuration.md:1477]:
```suggestion
Enter a comma-separated list of plugin identifiers to identify plugins to load even if they are unsigned. Plugins with modified signatures are never loaded.
```

File: docs/sources/administration/configuration.md [hunk 4]
```diff
@@ -1474,7 +1474,9 @@ Set to `true` if you want to test alpha plugins that are not yet ready for gener
...
+
+Please note that using this is _not_ recommended. For more information, refer to [Plugin signatures]({{< relref "../plugins/plugin-signatures.md" >}}).
```
Comment (achatterjee-grafana) [inline docs/sources/administration/configuration.md:1479]:
```suggestion
We do _not_ recommend using this option. For more information, refer to [Plugin signatures]({{< relref "../plugins/plugin-signatures.md" >}}).
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added doc change request.

**Review by achatterjee-grafana — APPROVED** [review]

**Comment by nodakai** [comment]
Few people would understand what's meant by "plugins with modified signatures are never loaded." At least I didn't.

It's important to use clear wording when communicating with the community, when the majority of them aren't "Grafana experts".  I'd write "If you have an invalid MANIFEST.txt in the plugin deployment directory, Grafana would not load the plugin even if `allow_loading_unsigned_plugin` is configured. Consider removing the manifest file altogether"

**Comment by wbrowne** [comment]
Thanks for your feedback @nodakai.

>If you have an invalid MANIFEST.txt

"Invalid" is not specific enough in this case. The signature `Modified` status specifically refers to when there is a content mismatch between what was found in the signature and what discovered when loading the plugin. For example if someone edited a plugin file after it was signed and tried to redistribute it as the original. The signatures statuses are described in [this docs section](https://grafana.com/docs/grafana/latest/administration/plugin-management/#plugin-signatures).

>Consider removing the manifest file altogether

This is not something that we recommend. The MANIFEST.txt is used to guarantee that a user is loading the plugin that the plugin author distributed.

If you think there are improvements to be made, you can open a pull request and we can go from there.

**Pull request merged**